homeassistant/components/config/config_entries.py

```diff
@@ -92,6 +93,27 @@ async def delete(self, request, entry_id):
         return self.json(result)
 
 
+class ConfigManagerEntryResourceReloadView(HomeAssistantView):
+    """View to reload a config entry."""
+
+    url = "/api/config/config_entries/entry/{entry_id}/reload"
+    name = "api:config:config_entries:entry:resource:reload"
+
+    async def post(self, request, entry_id):
+        """Reload a config entry."""
+        if not request["hass_user"].is_admin:
+            raise Unauthorized(config_entry_id=entry_id, permission="remove")
+
+        hass = request.app["hass"]
+
+        try:
+            result = await hass.config_entries.async_reload(entry_id)
+        except config_entries.UnknownEntry:
+            return self.json_message("Invalid entry specified", HTTP_NOT_FOUND)
+
+        return self.json({"require_restart": not result})
+
+
```

Comment:
We should update `entry_json` in this file to include if an entry is reloadable.

Reply:
an entry reload is just an unload followed by a load right? If we unload all when HA is shut down and load all when HA starts how can a reload not be possible? I'm not trying to hijack the review :) but I want to make sure I (and possibly others) understand this.